### DISCUSSION TITLE
How to reload read-only configuration without restarting the container

### DISCUSSION THREAD

**Discussion by jkjha** [oracle/opengrok#4278] [General]
I'm attempting to utilize the project grouping functionality within my Docker instance to group projects based on popularity and regular expressions. However, I've discovered that in order for changes to take effect on the UI, I must restart the container each time. This may be due to the fact that only during container deployment is the read-only.xml file merged with the actual configuration.xml file. I have searched the API documentation and other resources for an alternative approach, but have not found anything. Restarting the container does not seem like a practical solution.

As we are upgrading from v0.10.0 to the latest version, we are unfamiliar with many of the new features. I apologize for the numerous questions and would greatly appreciate any help you can provide.

**Comment by vladak** [comment]
Yes, this is how it works currently. Thinking of this, I am not sure whether it falls under the ['When not to use it'](https://github.com/oracle/opengrok/blob/master/docker/README.md#when-not-to-use-it) scenario for the Docker image. That said, the read-only configuration is present and documented, so it might make sense to have a way to reload it, however this enters a territory which I wanted to avoid in the main Docker image program and that is event handling and locking. The latter is needed because the persistent configuration is refreshed at the end of each indexing cycle which might collide with out-of-band event to merge the configuration with read-only config.
